src/frontend/change_buffer.rs

```diff
@@ -93,28 +110,63 @@ fn read_keys(
-                Some(IndexSelection::Single(i.saturating_sub(1)))
-            }
-            (IndexSelection::Single(i), KeyCode::Char('x')) => {
-                if i != (i + 1).clamp(0, max) {
-                    Some(IndexSelection::Range(i, usize::min(i + 1, max)))
-                } else {
-                    Some(IndexSelection::Single(i))
+
+            (IndexSelection::Single(i), KeyCode::Char(key)) => {
+                match resolve_command(&config, key) {
+                    Some(Command::Static(StaticCommand::MoveVisualLineDown)) => {
+                        Some(IndexSelection::Single(usize::min(i + 1, max)))
+                    }
+
+                    Some(Command::Static(StaticCommand::MoveVisualLineUp)) => {
+                        Some(IndexSelection::Single(usize::min(i.wrapping_sub(1), max)))
+                    }
+
+                    Some(Command::Static(StaticCommand::ExtendLineDown)) => {
+                        if i != (i + 1).clamp(0, max) {
+                            Some(IndexSelection::Range(i, usize::min(i + 1, max)))
+                        } else {
+                            Some(IndexSelection::Single(i))
+                        }
+                    }
+
+                    Some(Command::Static(StaticCommand::ExtendLineUp)) => {
+                        // TODO: I had a long day at work today - I have exactly zero clue what
+                        // should be written here.
+                        None
+                    }
+
+                    Some(Command::Static(StaticCommand::NoOp)) => None,
+
+                    None => None,
                 }
             }
-            (IndexSelection::Range(_, end), KeyCode::Char('j')) => {
-                Some(IndexSelection::Single(usize::min(end + 1, max)))
-            }
-            (IndexSelection::Range(start, _), KeyCode::Char('k')) => {
-                Some(IndexSelection::Single(start.saturating_sub(1)))
-            }
-            (IndexSelection::Range(start, end), KeyCode::Char('x')) => {
-                Some(IndexSelection::Range(start, usize::min(end + 1, max)))
+
+            (IndexSelection::Range(start, end), KeyCode::Char(key)) => {
+                match resolve_command(&config, key) {
+                    Some(Command::Static(StaticCommand::MoveVisualLineDown)) => {
+                        Some(IndexSelection::Single(usize::min(end + 1, max)))
+                    }
+
+                    Some(Command::Static(StaticCommand::MoveVisualLineUp)) => {
+                        Some(IndexSelection::Single(start.saturating_sub(1)))
+                    }
+
+                    Some(Command::Static(StaticCommand::ExtendLineDown)) => {
+                        Some(IndexSelection::Range(start, usize::min(end + 1, max)))
+                    }
+
+                    Some(Command::Static(StaticCommand::ExtendLineUp)) => {
+                        // TODO: I had a long day at work today - I have exactly zero clue what

```

Comment:
See above

Reply:
(unresolving, as apparently i forgot to fix this :skull:)